cmd/hydrate/src/lib.rs

```diff
@@ -0,0 +1,212 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! ## `humility hydrate`
+//!
+//! `humility hydrate` combines a raw memory dump (obtained from MGS) and a
+//! Hubris archive, forming a proper Hubris dump.
+//!
+//! ```console
+//! $ humility -a build-grapefruit-image-default.zip hydrate hubris.dry.0
+//! ```
+//!
+//! The archive is required, because the raw memory dump does not contain debug
+//! information.
+//!
+//! By default, this writes to `hubris.core.TASK_NAME.N` (where `N` is the
+//! lowest available value); use `--out` to specify a different path name.
+
+use anyhow::{anyhow, bail, Context, Result};
+use clap::{ArgGroup, IntoApp, Parser};
+use humility::hubris::HubrisFlashMap;
+use humility_arch_arm::ARMRegister;
+use humility_cli::{ExecutionContext, Subcommand};
+use humility_cmd::{Archive, Command, CommandKind};
+use humility_log::msg;
+use std::{collections::BTreeMap, io::Read, path::PathBuf};
+
+#[derive(Parser, Debug)]
+#[clap(
+    name = "hydrate", about = env!("CARGO_PKG_DESCRIPTION"),
+    group = ArgGroup::new("target").multiple(false)
+)]
+struct Args {
+    /// Path to write the resulting dump
+    #[clap(short, long)]
+    out: Option<PathBuf>,
+
+    /// Path to the raw memory dump
```

Comment:
turbo-nit: we _may_ want to add
```suggestion
    #[clap(short, long, value_hint = clap::ValueHint::FilePath)]
    out: Option<PathBuf>,

    /// Path to the raw memory dump
    #[clap(value_hint = clap::ValueHint::FilePath)]
```
to generate nicer completions, but...i don't believe we're currently generating completions for humility anyway, so it may not matter...

Reply:
I think our use of Clap is weird enough that completions would be a challenge, so I'm going to punt on this for now.